flowmachine/flowmachine/core/hour_slice.py

```diff
@@ -0,0 +1,258 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+from operator import lt as less_than, ge as greater_or_equal
+from typing import List, Union
+
+from sqlalchemy.sql import func, and_, or_, true
+
+__all__ = ["HourInterval", "HourSlice"]
+
+
+class DayPeriod:
+    """
+    Represents a repeated "daily" time period.
+    """
+
+    def __init__(self, weekday=None):
+        self.freq = "day"
+        self.weekday = weekday
+        if self.weekday is not None:
+            raise ValueError(
+                "If freq='day' then the `weekday` argument must not be provided."
+            )
+
+    def filter_timestamp_column_by_day_of_week(self, ts_col):
+        # no additional filtering needed to limit the day of the week
+        return true()
+
+
+class DayOfWeekPeriod:
+    """
+    Represents a repeated "day-of-the-week" time period.
+    """
+
+    valid_weekdays = [
+        "Monday",
+        "Tuesday",
+        "Wednesday",
+        "Thursday",
+        "Friday",
+        "Saturday",
+        "Sunday",
+    ]
+    weekday_indices_postgres = {
+        "Sunday": 0,
+        "Monday": 1,
+        "Tuesday": 2,
+        "Wednesday": 3,
+        "Thursday": 4,
+        "Friday": 5,
+        "Saturday": 6,
+    }
+
+    def __init__(self, weekday):
+        if weekday is None:
+            raise ValueError(
+                "If freq='week' then the `weekday` argument must be provided."
+            )
+        weekday = weekday.capitalize()
```

Comment:
How about `datetime.datetime.strptime(weekday, "%A").isoweekday()`?

Reply:
I like the idea, but isn't the `%A` locale-specific? What if someone has a non-English locale set?

Reply:
But that's good, no? Supports non-English locales, and you store it as the iso number of the weekday?